app.ts

```diff
@@ -0,0 +1,168 @@
+/** @format */
+
+class DataBase implements DataAggregator {
+    private static instance: DataBase;
+    private users: any[] = [];
+
+    public static getInstance(): DataBase {
+        if (!DataBase.instance) {
+            DataBase.instance = new DataBase();
+        }
+
+        return DataBase.instance;
+    }
+
+    public addUser(user: object): void {
+        this.users.push(user);
+    }
+
+    public getUsers(): object {
+        return this.users;
+    }
+
+    public removeUser(value: string, key: string): object {
+        this.users.filter(item => item[key] !== value)
+        return this.users;
+    }
+
+    public createIterator(): DataBaseIterator {
+        return new DataBaseIterator(this)
+    }
+}
+
+interface DataAggregator {
+    addUser(user: object): void;
+    getUsers(): object;
+    removeUser(value: string, key: string): object;
+    createIterator(): DataBaseIterator;
+}
+
+class DataBaseIterator {
+    private position: number = 0;
+
+    constructor(private users: DataAggregator) { }
+
+    public current(index: number): object {
```

Comment:
Why are we using an 'object' here? We can use the specific type our iterator is working with